src/databases/PlainText/avtPlainTextFileFormat.C

```diff
@@ -34,6 +34,16 @@
 using     std::string;
 using     std::vector;
 
+string
+TrimLeadingandTrailingWhitespace(string str)
+{
+    auto start = str.find_first_not_of(" \t");
```

Comment:
I've not seen `auto` used in quite this way before. I don't know any instances in VisIt source code now. Can you provide rationale? I am just curious.

Reply:
I used it purely because I didn't want to worry about what types were being passed around and thought I'd let the compiler sort it out instead. Also, I was curious if this was considered acceptable practice for our codebase, but I forgot to ask about it when I got distracted by the diff issues. 

After you asked me here, I did a bit of research and discovered that `auto` actually will increase compilation time, probably not by very much, but if I did this everywhere it could slow things down over time. I think I'll go ahead and change them to `size_t`.